src/radical/pilot/agent/launch_method/prte2.py

```diff
@@ -239,43 +241,35 @@ def _start_dvm(dvm_id, dvm_size, dvm_ready_flag):
                 num_slots = rm.cores_per_node * rm.smt
                 for node in node_list:
                     fout.write('%s slots=%d\n' % (node[0], num_slots))
-            vm_size = len(node_list)
 
-            dvm_ready = mt.Event()
-            dvm_ready_list.append(dvm_ready)
+            _dvm_size  = len(node_list)
+            _dvm_ready = mt.Event()
+            _dvm_uri   = _start_dvm(_dvm_id, _dvm_size, _dvm_ready)
 
-            partitions[str(_dvm_id)] = {
+            partition_id = str(_dvm_id)
+            partitions[partition_id] = {
                 'nodes'  : [node[1] for node in node_list],
-                'details': {'dvm_uri': _start_dvm(_dvm_id, vm_size, dvm_ready)}}
-
-        # check that all DVMs are ready
-        for _dvm_id, _dvm_ready in enumerate(dvm_ready_list):
-
-            if not _dvm_ready.is_set():
-
-                partition_id = str(_dvm_id)
+                'details': {'dvm_uri': _dvm_uri}}
```

Comment:
@andre-merzky with proposed changes not to keep nodes list per partition in `lm_info`, should this be updated as following?
```python
            partitions[partition_id] = {
                'nodes'  : [node[1] for node in node_list],
                'dvm_uri': _dvm_uri}
```
thus list of nodes will be moved to `ResourceManager.partitions` and the rest is as "details" per partition, which will be stored in `lm_info` as
```python
lm_info = {
    ...
    'partitions': {<partition_id>: {'dvm_uri': <dvm_uri>}}
    ...
}
```

Reply:
:+1: